File: external-libs/android-core/src/main/java/com/serhii/core/security/impl/crypto/CryptoSymmetric.java
package com.serhii.core.security.impl.crypto;

public interface CryptoSymmetric {

    Result encryptSymmetric(String message, final byte[] inputIV);

    Result decryptSymmetric(String message, final byte[] inputIV);

    Result encryptSymmetricWithKey(String message, String key, final byte[] inputIV);

    Result decryptSymmetricWithKey(String message, String key, final byte[] inputIV);
}

File: external-libs/android-core/src/main/java/com/serhii/core/security/impl/crypto/CryptoSymmetric.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.core.security.impl.crypto

interface CryptoSymmetric {
    fun encryptSymmetric(message: String, inputIV: ByteArray, key: String? = null): Result
    fun decryptSymmetric(message: String, inputIV: ByteArray, key: String? = null): Result
}